src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/directives/source_data.js

```diff
@@ -0,0 +1,42 @@
+import uiModules from 'ui/modules';
+import angular from 'angular';
+
+require('../styles/_source_data.less');
+
+const app = uiModules.get('kibana');
+
+app.directive('sourceData', function () {
+  return {
+    restrict: 'E',
+    scope: {
+      samples: '=',
+      sample: '='
+    },
+    template: require('../views/source_data.html'),
+    controller: function ($scope) {
+      const samples = $scope.samples;
+
+      if (samples.length > 0) {
+        $scope.selectedSample = samples[0];
+      }
+
+      $scope.$watch('selectedSample', (newValue) => {
+        $scope.sample = angular.copy(newValue);
```

Comment:
What's the reason for creating a copy? It doesn't seem like sample gets modified anywhere.